glean-core/src/histogram/mod.rs

```diff
@@ -136,3 +136,25 @@ impl<B: Bucketing> Histogram<B> {
         res
     }
 }
+
+impl<B> Histogram<B>
+where
+    B: Bucketing,
+    B: std::fmt::Debug,
+    B: PartialEq,
+{
+    /// Merges data from one histogram into the other.
+    ///
+    /// ## Panics
+    ///
+    /// Panics if the two histograms don't use the same bucketing.
```

Comment:
Really? Panics? That seems a little harsh. Though I suppose we're the only ones who should be making these, so we can ensure everything's okay.

Reply:
Sure. We control the use here. A histogram is an internal use. The only way to get there is through the buffered API and we make sure you can't mix that up.